pycqed/instrument_drivers/meta_instrument/LutMans/ro_lutman.py

```diff
@@ -262,105 +324,152 @@ def __init__(self, name, num_res: int=1, **kw):
-
-        FIXME: hardcode_case_0 is a workaround as long as it's not clear how
-        to make the qisa assembler output the codeword mask on DIO.
-        The first element of self.resonator_combinations is linked
-        to codeword 0, the others are not uploaded.
+        Load a single pulse to the lookuptable.
         '''
-        resonator_combinations = self.resonator_combinations()
-
-        pulse_type = self.pulse_type()
-        if regenerate_waveforms:
-            wave_dict = self.generate_standard_waveforms()
-        else:
-            wave_dict = self._wave_dict
-        I_waves = []
-        Q_waves = []
-        cases = np.zeros([len(resonator_combinations)])
-
-        for i, resonator_combination in enumerate(resonator_combinations):
-            if not resonator_combination:
-                # empty combination (= []), generating empty 20 ns pulse
-                I_waves.append(np.zeros(36))
-                Q_waves.append(np.zeros(36))
-            else:
-                for j, resonator in enumerate(resonator_combination):
-                    wavename = pulse_type+'_R'+str(resonator)
-                    if j == 0:
-                        # starting the entry
-                        Icopy = copy.deepcopy(wave_dict[wavename][0])
-                        Qcopy = copy.deepcopy(wave_dict[wavename][1])
-                        I_waves.append(Icopy)
-                        Q_waves.append(Qcopy)
-                    else:
-                        # adding new wave (not necessarily same length)
-                        I_waves[i] = add_waves_different_length(
-                            I_waves[i], wave_dict[wavename][0])
-                        Q_waves[i] = add_waves_different_length(
-                            Q_waves[i], wave_dict[wavename][1])
-                    cases[i] += 2**self._resonator_codeword_bit_mapping.index(resonator)
-
-            # clipping the waveform
-            I_waves[i] = np.clip(I_waves[i],
-                                 self._voltage_min, self._voltage_max)
-            Q_waves[i] = np.clip(Q_waves[i], self._voltage_min,
-                                 self._voltage_max)
-
-        self.AWG.get_instr().awg_sequence_acquisition_and_DIO_triggered_pulse(
-            I_waves, Q_waves, cases, self.acquisition_delay(), timeout=timeout)
+        self._mode = 'DIO_triggered'
+        self.timeout(timeout)
+        self.load_waveforms_onto_AWG_lookuptable(regenerate_waveforms=regenerate_waveforms)
 
     def set_mixer_offsets(self):
         UHFQC = self.AWG.get_instr()
         UHFQC.sigouts_0_offset(self.mixer_offs_I())
         UHFQC.sigouts_1_offset(self.mixer_offs_Q())
 
-    def load_waveforms_onto_AWG_lookuptable(
-            self, regenerate_waveforms: bool=True,
-            stop_start: bool = True):
-        raise NotImplementedError(
-            'UHFQC needs a full sequence, use '
-            '"load_DIO_triggered_sequence_onto_UHFQC"')
+    def _program_hash_differs(self)-> bool:
```

Comment:
Should we not allows this function to disappear? Given that uploading is done in the UHFQA now?

Reply:
@AdriaanRol If you are happy with getting rid of the two operating modes of the LutMan then yes, absolutely. I don't have insight into how these functions are used in your experiments, so I didn't dare to touch them.